### DISCUSSION TITLE
Hosting Setups

### DISCUSSION THREAD

**Discussion by apocist** [yogstation13/Yogstation#10229] [Help - Hosting]
Does anyone have any recommended setups, scripts, Dockers, guides, or such to get a testing server up and running from scratch (following the wiki). Along with the various compile issues, getting someone to work on something like Linode would be amazing.

https://wiki.yogstation.net/wiki/Downloading_the_source_code#Hosting_a_server

edit: updated with link to Wiki references

**Comment by alexkar598** [comment]
If youre using linux you have two options:
1) Use TGS to manage the server for you (Production Grade stuff, overkill for local)
2) Host it yourself using BYOND directly, I made a small guide for this option [here](https://gist.github.com/alexkar598/9a5624decad39ce40c5dadef8231fbd3)
3) Technically theres a Dockerfile in the repo but its outdated and I don't think it works anymore

For windows:
1) You can also use TGS which works on windows
2) You can just use dreamdaemon directly

We have a few TGS scripts we use on the server to install dependencies and to build and publish a package for the resource files to deploy on our "CDN"(cloudflare with caching page rules)

**Comment by AshCorr** [comment]
TGS also has a Docker image which is what we use in Yog's production environment.

https://hub.docker.com/r/tgstation/server/
